service/dynamodb/expression/condition.go

```diff
@@ -0,0 +1,225 @@
+}
+
+// Equal will create a ConditionBuilder. This will be the method for
+// ValueBuilder
+func (v ValueBuilder) Equal(right OperandBuilder) ConditionBuilder {
+	return Equal(v, right)
+}
+
+// Equal will create a ConditionBuilder. This will be the method for SizeBuilder
+func (s SizeBuilder) Equal(right OperandBuilder) ConditionBuilder {
+	return Equal(s, right)
+}
+
+// And will create a ConditionBuilder. This will be the function call
+func And(cond ...ConditionBuilder) ConditionBuilder {
+	return ConditionBuilder{
+		conditionList: cond,
+		Mode:          AndCond,
+	}
+}
+
+// And will create a ConditionBuilder. This will be the method signature
+func (cond ConditionBuilder) And(right ...ConditionBuilder) ConditionBuilder {
+	right = append(right, cond)
+	return And(right...)
+}
+
+// Between will create a ConditionBuilder. This will be the function signature.
+func Between(ope, lower, upper OperandBuilder) ConditionBuilder {
+	return ConditionBuilder{
+		operandList: []OperandBuilder{ope, lower, upper},
+		Mode:        BetweenCond,
+	}
+}
+
+// Between will create a ConditionBuilder. This will be the method signature for
+// PathBuilders. There will be a check for the amount of OperandBuilders later
+// (only should be three)
+func (p PathBuilder) Between(lower, upper OperandBuilder) ConditionBuilder {
+	return Between(p, lower, upper)
+}
+
+// Between will create a ConditionBuilder. This will be the method signature for
+// ValueBuilders. There will be a check for the amount of OperandBuilders later
+// (only should be three)
+func (v ValueBuilder) Between(lower, upper OperandBuilder) ConditionBuilder {
+	return Between(v, lower, upper)
+}
+
+// Between will create a ConditionBuilder. This will be the method signature for
+// SizeBuilders. There will be a check for the amount of OperandBuilders later
+// (only should be three)
+func (s SizeBuilder) Between(lower, upper OperandBuilder) ConditionBuilder {
+	return Between(s, lower, upper)
+}
+
+// BuildExpression will take an ConditionBuilder as input and output an
+// Expression
+func (cond ConditionBuilder) BuildExpression() (Expression, error) {
+	en, err := cond.buildCondition()
+	if err != nil {
+		return Expression{}, err
+	}
+
+	expr, err := en.buildExprNodes(&aliasList{})
+	if err != nil {
+		return Expression{}, err
+	}
+
+	return expr, nil
+}
+
+// buildCondition will iterate over the tree of ConditionBuilders and
+// OperandBuilders and build a tree of ExprNodes
+func (cond ConditionBuilder) buildCondition() (ExprNode, error) {
+	switch cond.Mode {
+	case EqualCond:
+		return compareBuildCondition(cond)
+	case AndCond:
+		return boolBuildCondition(cond)
+	case BetweenCond:
+		return betweenBuildCondition(cond)
+	}
+	return ExprNode{}, fmt.Errorf("No matching Mode to %v", cond.Mode)
+}
+
+// compareBuildCondition is the function to make ExprNodes from Compare
+// ConditionBuilders
+func compareBuildCondition(c ConditionBuilder) (ExprNode, error) {
+	if len(c.conditionList) != 0 {
```

Comment:
Move these checks into a simple validate method which returns an error. This will remove some duplicate lines